bokeh/command/bootstrap.py

```diff
@@ -96,6 +99,15 @@ def main(argv):
         prog=argv[0],
         epilog="See '<command> --help' to read about a specific subcommand.")
 
+    if sys.platform == 'win32':
+        """
+        windows hack
+        """
+        argv0_ext = os.path.splitext(argv[0])[-1]
+        if argv0_ext != '.py':
```

Comment:
The comment should be a comment (rather than an string) and it would be good to reference the actual issue this clunky code is for.  But also this logic seems like it would not work well with directory style apps? I'm much less inclined to resort to a tacky solution if it only works part of the time, in that case I'd rather just document the necessary invocation for windows users and move on.

Reply:
I'll change my pull request for a doc entry.
Actually it's a hack to work around a tornado issue
When invoking the autoreload with bokeh or bokeh.exe, on tornado side it tries to reconstruct cmd invocation but it ends by calling :
`python.exe -m bokeh.[exe] serve ... `
Which is not a valid command

Reply:
Where should I put the doc?
Directly in the __init__.py of commands?
However it does not seems to be referenced by the .rst

Reply:
@xavArtley I think a note should definitely go here:

https://github.com/bokeh/bokeh/blob/master/bokeh/command/subcommands/serve.py#L518

And, as long as you are touching that, can you fix up the broken missing spaces between sentences? I didn't realize those help strings used implicit string concat (I don't like that personally but if that's what will be used then spaces need to be inserted appropriately). The text in the corresponding test file will need to be updated as well. 

The docs there will automatically be added to the reference guide too, tho I don't know that those are used much. If you want to also add some narrative docs in User's Guide, then I think a new small section for "dev mode" would be appropriate here:

https://raw.githubusercontent.com/bokeh/bokeh/master/sphinx/source/docs/user_guide/server.rst

As an aside `server.rst` is getting big enough that we could probably consider splitting it up but that can happen separately.